src/core/server/http/integration_tests/__fixtures__/plugins/dummy_security/server/plugin.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import Boom from 'boom';
+import { Authenticate, CoreSetup } from '../../../../../../../../core/server';
+
+interface User {
+  id: string;
+  roles?: string[];
+}
+
+interface Storage {
+  value: User;
+  expires: number;
+}
+
+export class DummySecurityPlugin {
+  public setup(core: CoreSetup) {
+    const authenticate: Authenticate<Storage> = async (request, sessionStorage, t) => {
+      if (request.path === '/auth/has_session') {
+        const prevSession = await sessionStorage.get();
+        const userData = prevSession.value;
+        sessionStorage.set({ value: userData, expires: Date.now() + 1000 });
+
+        return t.authenticated({ credentials: userData });
+      }
+
+      if (request.headers.authorization) {
+        const user = { id: '42' };
+        sessionStorage.set({ value: user, expires: Date.now() + 1000 });
+        return t.authenticated({ credentials: user });
+      } else {
+        return t.rejected(Boom.unauthorized());
```

Comment:
are we going to ban `Boom` usage? there is an issue, but don't see any discussions https://github.com/elastic/kibana/issues/12947